### PR TITLE
[AMORO-2477] [Improvement]: Amoro Flink Optimizer Support Flink 1.18

### LINKED ISSUE #2477: [Improvement]: Amoro Flink Optimizer Support Flink 1.18
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/amoro/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

Amoro Flink Optimizer Support Flink 1.18 And Supports using -d parameter when compiling,such as `mvn clean package -pl 'ams/optimizer/flink-optimizer' -Doptimizer-flink.version=1.18.0`

### How should we improve?

- Amoro Flink Optimizer Pom Default Flink Version change to 1.18
- Remove Flink 1.14 Flink Optimizer Support,Minimum Flink Optimizer supported version is 1.15

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by czy006**
## Why are the changes needed?

Close #2477.

## Brief change log

- Amoro Flink Optimizer Support Flink 1.18
- Use Flink 1.18 to default Amoro Flink Optimizer Version
- Remove Flink 1.14 Flink Optimizer Support,Minimum Flink Optimizer supported version is 1.15

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible
- [ ] Add screenshots for manual tests if appropriate
- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (no)
- If yes, how is the feature documented? (not documented)

### CODE REVIEW

**Comment by czy006** [comment]
`mvn clean package -Dflink-optimizer.flink-version=1.17.2`

**Review by baiyangtx — COMMENTED** [review]
It seems that the code of Flink Optimizer is completely independent of the Flink version. 

Do we still need to maintain different Flink versions in the profile?

 Can we just directly depend on a newer Flink version for compilation and indicate the minimum supported Flink version in the version dependency matrix?

**Comment by czy006** [comment]
> It seems that the code of Flink Optimizer is completely independent of the Flink version.
> 
> Do we still need to maintain different Flink versions in the profile?
> 
> Can we just directly depend on a newer Flink version for compilation and indicate the minimum supported Flink version in the version dependency matrix?

Yes, I agree that using a higher version of Flink by default is a wiser choice. I'll correct it later

File: ams/optimizer/flink-optimizer/pom.xml [hunk 1]
```diff
@@ -33,8 +33,9 @@
     <url>https://amoro.netease.com</url>
 
     <properties>
-        <optimizer-flink.version>1.14.6</optimizer-flink.version>
-        <flink.scala.suffix>_2.12</flink.scala.suffix>
+        <flink-optimizer.flink-version>1.18.1</flink-optimizer.flink-version>
+        <flink-optimizer.scala-version>2.12</flink-optimizer.scala-version>
+        <flink-optimizer.scala-version_suffix></flink-optimizer.scala-version_suffix>
```
Comment (zhoujinsong) [inline ams/optimizer/flink-optimizer/pom.xml:38]: The empty property value may be confusing, how about adding some comment here to tell why?

**Review by zhoujinsong — COMMENTED** [review]
This PR changed some property names, we may need to adapt to this change in docs, image configuration files and README.md.

File: README.md [hunk 2]
```diff
@@ -135,8 +135,8 @@ Amoro is built using Maven with Java 1.8 and Java 17(only for `mixed/trino` modu
 * To skip tests: `mvn -DskipTests package -P toolchain`
 * To package without trino module and JAVA 17 dependency: `mvn clean package -DskipTests -pl '!mixed/trino'`
 * To build with hadoop 2.x(the default is 3.x) `mvn clean package -DskipTests -Dhadoop=v2`
-* To indicate flink version for optimizer(the default is 1.14, 1.15 and 1.16 are available)
-  `mvn clean package -DskipTests -Doptimizer.flink=1.15`
+* To indicate flink version for optimizer(the default is 1.18.1,example for compiled is lower than 1.15)
```
Comment (zhoujinsong) [inline README.md:138]:
```suggestion
* To indicate Flink version for optimizer (the default is 1.18.1): `mvn clean package -Dflink-optimizer.flink-version=1.15.4`. If the version of Flink is below 1.5.0, you also need to add the `-Pflink-pre-1.15` parameter: `mvn clean package -Pflink-pre-1.15 -Dflink-optimizer.flink-version=1.14.6`.
```

**Review by zhoujinsong — COMMENTED** [review]
@czy006 I left a small suggestion about the doc, please take another look.

**Review by zhoujinsong — APPROVED** [review]
LGTM.
Thanks a lot for your contribution!

**Pull request merged**